library/Icingadb/Data/Nodes.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Icinga\Module\Icingadb\Data;
+
+use ArrayIterator;
+use Countable;
+use Icinga\Module\Icingadb\Common\Auth;
+use Icinga\Module\Icingadb\Common\Database;
+use Icinga\Module\Icingadb\Model\DependencyNode;
+use Icinga\Module\Icingadb\Model\Host;
+use ipl\Stdlib\Filter;
+use IteratorAggregate;
+
+class Nodes implements IteratorAggregate, Countable
```

Comment:
`DependencyNodes`?

And, please document in the class docstring why this is needed.
And add an `@internal` note prohibiting usage outside Icinga DB Web